### PR TITLE
Improve automation

### PR DESCRIPTION
**Pull request by nunnatsa**
1. use pre-built bundle to deploy HCO
2. create several VMs and run them, to check must-gather

```release-note
None
```

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

### CODE REVIEW

File: automation/deploy_hco.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    name: community-kubevirt-hyperconverged
+    channel: "${HCO_VERSION}"
+EOF
+
+# Wait for the deployments to be available
+"${K_COMMAND}" wait -n "${NS}" deployment hco-operator --for=condition=Available --timeout="1080s"
+"${K_COMMAND}" wait -n "${NS}"  deployment hco-webhook --for=condition=Available --timeout="1080s"
+
+# Create the HyperConverged resource
+curl "https://raw.githubusercontent.com/kubevirt/hyperconverged-cluster-operator/main/deploy/hco.cr.yaml" | ${K_COMMAND} -n "${NS}" apply -f -
```
Comment (orenc1) [inline automation/deploy_hco.sh:59]: you use HCO version 1.4.0, but here deploy the CR from master branch, not release-1.4

Comment (nunnatsa) [reply automation/deploy_hco.sh]: I think it's a low risk. unknown fields are ignored while maintaining two files in two different places is also risky.

File: automation/deploy_hco.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+HCO_VERSION=1.4.0
+CMD=oc
+NS=kubevirt-hyperconverged
+
+# Create a catalog source
+cat <<EOF | ${CMD} apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: hco-unstable-catalog-source
```
Comment (orenc1) [inline automation/deploy_hco.sh:14]: according to the value of `${HCO_VERSION}`, this does not seem to be the unstable version.

File: automation/deploy_hco.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: "quay.io/kubevirt/hyperconverged-cluster-index:${HCO_VERSION}"
```
Comment (orenc1) [inline automation/deploy_hco.sh:18]: quotes are not needed in image pullspec

File: automation/deploy_hco.sh [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  displayName: Kubevirt Hyperconverged Cluster Operator
+  publisher: Kubevirt Project
+EOF
+
+# Create the kubevirt-hyperconverged namespace
+cat <<EOF | ${CMD} apply -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+    name: "${NS}"
```
Comment (orenc1) [inline automation/deploy_hco.sh:28]: same for namespace name. anyway space is not allowed in kubernetes object name.

File: automation/deploy_hco.sh [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+# Create Subscription
+cat <<EOF | ${CMD} apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+    name: hco-operatorhub
+    namespace: "${NS}"
+spec:
+    source: hco-unstable-catalog-source
```
Comment (orenc1) [inline automation/deploy_hco.sh:48]: remove `unstable`

File: automation/deploy_hco.sh (hunk 3 above)
Comment (nunnatsa) [reply automation/deploy_hco.sh:18]: I prefer to use them when I'm using environment variables. Anyway, it's working and I don't see why it's worng.

File: automation/deploy_hco.sh (hunk 4 above)
Comment (nunnatsa) [reply automation/deploy_hco.sh:28]: same - why do we care?

File: automation/deploy_hco.sh (hunk 5 above)
Comment (nunnatsa) [reply automation/deploy_hco.sh]: fixed

File: automation/deploy_hco.sh (hunk 2 above)
Comment (nunnatsa) [reply automation/deploy_hco.sh]: fixed

File: automation/create_workloads.sh [hunk 6]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+set -ex
+
+CMD=oc
+
+for ns in {1..5}; do
+  ${CMD} create namespace "ns${ns}"
+  sed "s#__NS__#${ns}#"g automation/vm.yaml | ${CMD} apply -f -
+  ./virtctl start -n "ns${ns}" "testvm${ns}"
```
Comment (orenc1) [inline automation/create_workloads.sh:10]: you can define the VMs [here](https://github.com/kubevirt/must-gather/pull/84/files#diff-0b490f49df7ff9face7e12ba4004b5fc963316fb7a9754f4f0ffea591a4c0046R7) to be `running: true` and they will start up (and VMIs will be created) on definition, and you won't need to use `virtctl start`

Comment (nunnatsa) [reply automation/create_workloads.sh:10]: Maybe I'll use it in the future.

**Review by orenc1 — APPROVED** [review]
/lgtm
/release-note-none

File: automation/deploy_hco.sh [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    name: community-kubevirt-hyperconverged
+    channel: "${HCO_VERSION}"
+EOF
+
+# Wait for the deployments to be available
+"${CMD}" wait -n "${NS}" deployment hco-operator --for=condition=Available --timeout="1080s"
+"${CMD}" wait -n "${NS}"  deployment hco-webhook --for=condition=Available --timeout="1080s"
+
+# Create the HyperConverged resource
+curl "https://raw.githubusercontent.com/kubevirt/hyperconverged-cluster-operator/release-1.4/deploy/hco.cr.yaml" | ${CMD} -n "${NS}" apply -f -
```
Comment (masayag) [inline automation/deploy_hco.sh:59]: HCO_VERSION is a variable set to 1.4.0, but here the link to the CR is hard-coded.
For sake of consistency and not missing this one when version is promoted, would you prefer to extract `1.4` into a variable as well ?

Comment (orenc1) [reply automation/deploy_hco.sh:59]: I agree with Moti.
Something like
```
RELEASE_BRANCH=release-${HCO_VERSION%.*}
```
and
```
 curl "https://raw.githubusercontent.com/kubevirt/hyperconverged-cluster-operator/${RELEASE_BRANCH}/deploy/hco.cr.yaml" | ${CMD} -n "${NS}" apply -f -

Comment (nunnatsa) [reply automation/deploy_hco.sh]: Done + add DVs

**Review by orenc1 — APPROVED** [review]
/lgtm

**Pull request merged**